switchlink/switchlink_main.c

```diff
@@ -288,6 +289,13 @@ static void nl_process_event_loop(void) {
 
     ret = select(num_fds, &read_fds, NULL, NULL, NULL);
     if (ret == -1) {
+      // Select system call being interrupted by a signal EINTR.
+      // This is not really an error condition, hence continue the loop.
+      if (errno == EINTR)
+        continue;
+
+      krnlmon_log_error("Received select error, errno :%d, meaning: %s",
+                         errno, strerror(errno));
       perror("pselect");
       return;
     } else if (ret == 0) {
```

Comment:
According to the `select` documentation, "The return value may be zero if the timeout expired before any file descriptors became ready."

We don't supply a timeout argument to `select`. Can this condition even occur?

Would a brief explanation comment be in order here?

Reply:
Right, we wont hit a condition where "ret == 0", I can remove that. Right now we are not doing anything when ret == 0, it just continues the loop.